<a href="https://colab.research.google.com/github/mbparsons/code/blob/master/Scraping_Instagram_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importing Data

In [0]:
!pip install instagram-scraper 

  Created wheel for instagram-scraper: filename=instagram_scraper-1.8.1-cp36-none-any.whl size=34275 sha256=a60abd64d174514227bb66ec66253afd08ec957443a0e7f22ae57e38e2418b61
  Stored in directory: /root/.cache/pip/wheels/b6/53/4e/67f0a88142d60eac158eb38df3c7a159cc37358d638913a183
Successfully built instagram-scraper


In [0]:
from instagram_scraper import InstagramScraper

In [0]:
from google.colab import drive
drive.mount('/content/drive')

In [0]:
!instagram-scraper ford --media-metadata -m 200 -d 'drive/My Drive/Colab Notebooks/Instagram/'

In [0]:
!instagram-scraper volvocars --media-metadata -m 200 -d 'drive/My Drive/Colab Notebooks/Instagram/'

In [0]:
!instagram-scraper dodgeofficial --media-metadata -m 200 -d 'drive/My Drive/Colab Notebooks/Instagram/'

In [0]:
!instagram-scraper chevrolet --media-metadata -m 200 -d 'drive/My Drive/Colab Notebooks/Instagram/'

In [0]:
!instagram-scraper honda --media-metadata -m 200 -d 'drive/My Drive/Colab Notebooks/Instagram/'

In [0]:
!instagram-scraper ford --profile-metadata -d 'drive/My Drive/Colab Notebooks/Instagram/Profiles'

In [0]:
!instagram-scraper volvocars --profile-metadata -d 'drive/My Drive/Colab Notebooks/Instagram/Profiles'

In [0]:
!instagram-scraper dodgeofficial --profile-metadata -d 'drive/My Drive/Colab Notebooks/Instagram/Profiles'

In [0]:
!instagram-scraper chevrolet --profile-metadata -d 'drive/My Drive/Colab Notebooks/Instagram/Profiles'

In [0]:
!instagram-scraper honda --profile-metadata -d 'drive/My Drive/Colab Notebooks/Instagram/Profiles'

In [0]:
pathprofile1 = 'drive/My Drive/Colab Notebooks/Instagram/Profiles/ford.json'
fordprofile = open(pathprofile1, 'r')
fordprofile = open(pathprofile1, 'r')

In [0]:
pathprofile2 = 'drive/My Drive/Colab Notebooks/Instagram/Profiles/volvocars.json'
volvoprofile = open(pathprofile2, 'r')
parsedjsonvolvoprofile = json.load(volvoprofile)

In [0]:
pathprofile3 = 'drive/My Drive/Colab Notebooks/Instagram/Profiles/dodgeofficial.json'
dodgeprofile = open(pathprofile3, 'r')
parsedjsondodgeprofile = json.load(dodgeprofile)

In [0]:
pathprofile4 = 'drive/My Drive/Colab Notebooks/Instagram/Profiles/chevrolet.json'
chevroletprofile = open(pathprofile4, 'r')
parsedjsonchevroletprofile = json.load(chevroletprofile)

In [0]:
pathprofile5 = 'drive/My Drive/Colab Notebooks/Instagram/Profiles/honda.json'
hondaprofile = open(pathprofile5, 'r')
parsedjsonhondaprofile = json.load(hondaprofile)

In [0]:
import glob
import json
import datetime
import csv
import pickle

In [0]:
path1 = 'drive/My Drive/Colab Notebooks/Instagram/ford.json'
fordjson = open(path1, 'r')
parsedjsonford = json.load(fordjson)

In [0]:
path2 = 'drive/My Drive/Colab Notebooks/Instagram/volvocars.json'
volvojson = open(path2, 'r')
parsedjsonvolvo = json.load(volvojson)

In [0]:
path3 = 'drive/My Drive/Colab Notebooks/Instagram/dodgeofficial.json'
dodgejson = open(path3, 'r')
parsedjsondodge= json.load(dodgejson)

In [0]:
path4 = 'drive/My Drive/Colab Notebooks/Instagram/chevrolet.json'
chevroletjson = open(path4, 'r')
parsedjsonchevrolet= json.load(chevroletjson)

In [0]:
path5 = 'drive/My Drive/Colab Notebooks/Instagram/honda.json'
hondajson = open(path5, 'r')
parsedjsonhonda= json.load(hondajson)

In [0]:
jsonfiles = glob.glob('*/*.json')

In [0]:
for afile in jsonfiles:
    print(afile)
    outputfile = open('drive/My Drive/Colab Notebooks/Instagram/ford.csv', 'w')
     
    csvwriter = csv.writer(outputfile)
  
    instagramposts = {}
     
    metadata = {}
    metadata['is_video'] = "Photo"
    metadata['display_url'] = ""
    metadata['caption'] = ""
    metadata['like count'] = ""
    metadata['comment count'] = ""
    metadata['tags'] = ""
    metadata['time'] = ""
    header = list(metadata.keys())
    csvwriter.writerows([header])
     
    for apost in parsedjsonford['GraphImages']:    
 
        metadata = {}
        metadata['is_video'] = "Photo"
        metadata['display_url'] = ""
        metadata['caption'] = ""
        metadata['like count'] = ""
        metadata['comment count'] = ""
        metadata['tags'] = ""
        metadata['time'] = ""
         
        if 'edge_media_to_caption' in apost:
            if 'edges' in apost['edge_media_to_caption']:
                edges = apost['edge_media_to_caption']['edges']
                if len(edges) > 0:
    #                    print("woo")
                    metadata['caption'] = apost['edge_media_to_caption']['edges'][0]['node']['text']
            if 'edge_media_preview_like' in apost:
                if 'count' in apost['edge_media_preview_like']:
                    metadata['like count'] = apost['edge_media_preview_like']['count']
            if 'edge_media_to_comment' in apost:
                if 'count' in apost['edge_media_to_comment']:
                    metadata['comment count'] = apost['edge_media_to_comment']['count']
            if 'tags' in apost:
                metadata['tags'] = apost['tags']
                     
            metadata['time'] = datetime.datetime.fromtimestamp(int(apost['taken_at_timestamp'])).strftime('%Y-%m-%d %H:%M')
            metadata['display_url'] = apost['display_url']
            if apost['is_video'] == True:
                metadata['is_video'] = "Video"
      
            row = []
            for anitem in metadata:
                row.append(metadata[anitem])
            csvwriter.writerow(row)
             
    outputfile.close()

In [0]:
for afile in jsonfiles:
    print(afile)
    outputfile = open('drive/My Drive/Colab Notebooks/Instagram/volvo.csv', 'w')
     
    csvwriter = csv.writer(outputfile)
  
    instagramposts = {}
     
    metadata = {}
    metadata['is_video'] = "Photo"
    metadata['display_url'] = ""
    metadata['caption'] = ""
    metadata['like count'] = ""
    metadata['comment count'] = ""
    metadata['tags'] = ""
    metadata['time'] = ""
    header = list(metadata.keys())
    csvwriter.writerows([header])
     
    for apost in parsedjsonvolvo['GraphImages']:    
 
        metadata = {}
        metadata['is_video'] = "Photo"
        metadata['display_url'] = ""
        metadata['caption'] = ""
        metadata['like count'] = ""
        metadata['comment count'] = ""
        metadata['tags'] = ""
        metadata['time'] = ""
         
        if 'edge_media_to_caption' in apost:
            if 'edges' in apost['edge_media_to_caption']:
                edges = apost['edge_media_to_caption']['edges']
                if len(edges) > 0:
    #                    print("woo")
                    metadata['caption'] = apost['edge_media_to_caption']['edges'][0]['node']['text']
            if 'edge_media_preview_like' in apost:
                if 'count' in apost['edge_media_preview_like']:
                    metadata['like count'] = apost['edge_media_preview_like']['count']
            if 'edge_media_to_comment' in apost:
                if 'count' in apost['edge_media_to_comment']:
                    metadata['comment count'] = apost['edge_media_to_comment']['count']
            if 'tags' in apost:
                metadata['tags'] = apost['tags']
                     
            metadata['time'] = datetime.datetime.fromtimestamp(int(apost['taken_at_timestamp'])).strftime('%Y-%m-%d %H:%M')
            metadata['display_url'] = apost['display_url']
            if apost['is_video'] == True:
                metadata['is_video'] = "Video"
      
            row = []
            for anitem in metadata:
                row.append(metadata[anitem])
            csvwriter.writerow(row)
             
    outputfile.close()

In [0]:
for afile in jsonfiles:
    print(afile)
    outputfile = open('drive/My Drive/Colab Notebooks/Instagram/dodgeofficial.csv', 'w')
     
    csvwriter = csv.writer(outputfile)
  
    instagramposts = {}
     
    metadata = {}
    metadata['is_video'] = "Photo"
    metadata['display_url'] = ""
    metadata['caption'] = ""
    metadata['like count'] = ""
    metadata['comment count'] = ""
    metadata['tags'] = ""
    metadata['time'] = ""
    header = list(metadata.keys())
    csvwriter.writerows([header])
     
    for apost in parsedjsondodge['GraphImages']:    
 
        metadata = {}
        metadata['is_video'] = "Photo"
        metadata['display_url'] = ""
        metadata['caption'] = ""
        metadata['like count'] = ""
        metadata['comment count'] = ""
        metadata['tags'] = ""
        metadata['time'] = ""
         
        if 'edge_media_to_caption' in apost:
            if 'edges' in apost['edge_media_to_caption']:
                edges = apost['edge_media_to_caption']['edges']
                if len(edges) > 0:
    #                    print("woo")
                    metadata['caption'] = apost['edge_media_to_caption']['edges'][0]['node']['text']
            if 'edge_media_preview_like' in apost:
                if 'count' in apost['edge_media_preview_like']:
                    metadata['like count'] = apost['edge_media_preview_like']['count']
            if 'edge_media_to_comment' in apost:
                if 'count' in apost['edge_media_to_comment']:
                    metadata['comment count'] = apost['edge_media_to_comment']['count']
            if 'tags' in apost:
                metadata['tags'] = apost['tags']
                     
            metadata['time'] = datetime.datetime.fromtimestamp(int(apost['taken_at_timestamp'])).strftime('%Y-%m-%d %H:%M')
            metadata['display_url'] = apost['display_url']
            if apost['is_video'] == True:
                metadata['is_video'] = "Video"
      
            row = []
            for anitem in metadata:
                row.append(metadata[anitem])
            csvwriter.writerow(row)
             
    outputfile.close()

In [0]:
for afile in jsonfiles:
    print(afile)
    outputfile = open('drive/My Drive/Colab Notebooks/Instagram/chevrolet.csv', 'w')
     
    csvwriter = csv.writer(outputfile)
  
    instagramposts = {}
     
    metadata = {}
    metadata['is_video'] = "Photo"
    metadata['display_url'] = ""
    metadata['caption'] = ""
    metadata['like count'] = ""
    metadata['comment count'] = ""
    metadata['tags'] = ""
    metadata['time'] = ""
    header = list(metadata.keys())
    csvwriter.writerows([header])
     
    for apost in parsedjsonchevrolet['GraphImages']:    
 
        metadata = {}
        metadata['is_video'] = "Photo"
        metadata['display_url'] = ""
        metadata['caption'] = ""
        metadata['like count'] = ""
        metadata['comment count'] = ""
        metadata['tags'] = ""
        metadata['time'] = ""
         
        if 'edge_media_to_caption' in apost:
            if 'edges' in apost['edge_media_to_caption']:
                edges = apost['edge_media_to_caption']['edges']
                if len(edges) > 0:
    #                    print("woo")
                    metadata['caption'] = apost['edge_media_to_caption']['edges'][0]['node']['text']
            if 'edge_media_preview_like' in apost:
                if 'count' in apost['edge_media_preview_like']:
                    metadata['like count'] = apost['edge_media_preview_like']['count']
            if 'edge_media_to_comment' in apost:
                if 'count' in apost['edge_media_to_comment']:
                    metadata['comment count'] = apost['edge_media_to_comment']['count']
            if 'tags' in apost:
                metadata['tags'] = apost['tags']
                     
            metadata['time'] = datetime.datetime.fromtimestamp(int(apost['taken_at_timestamp'])).strftime('%Y-%m-%d %H:%M')
            metadata['display_url'] = apost['display_url']
            if apost['is_video'] == True:
                metadata['is_video'] = "Video"
      
            row = []
            for anitem in metadata:
                row.append(metadata[anitem])
            csvwriter.writerow(row)
             
    outputfile.close()

In [0]:
for afile in jsonfiles:
    print(afile)
    outputfile = open('drive/My Drive/Colab Notebooks/Instagram/honda.csv', 'w')
     
    csvwriter = csv.writer(outputfile)
  
    instagramposts = {}
     
    metadata = {}
    metadata['is_video'] = "Photo"
    metadata['display_url'] = ""
    metadata['caption'] = ""
    metadata['like count'] = ""
    metadata['comment count'] = ""
    metadata['tags'] = ""
    metadata['time'] = ""
    header = list(metadata.keys())
    csvwriter.writerows([header])
     
    for apost in parsedjsonhonda['GraphImages']:    
 
        metadata = {}
        metadata['is_video'] = "Photo"
        metadata['display_url'] = ""
        metadata['caption'] = ""
        metadata['like count'] = ""
        metadata['comment count'] = ""
        metadata['tags'] = ""
        metadata['time'] = ""
         
        if 'edge_media_to_caption' in apost:
            if 'edges' in apost['edge_media_to_caption']:
                edges = apost['edge_media_to_caption']['edges']
                if len(edges) > 0:
    #                    print("woo")
                    metadata['caption'] = apost['edge_media_to_caption']['edges'][0]['node']['text']
            if 'edge_media_preview_like' in apost:
                if 'count' in apost['edge_media_preview_like']:
                    metadata['like count'] = apost['edge_media_preview_like']['count']
            if 'edge_media_to_comment' in apost:
                if 'count' in apost['edge_media_to_comment']:
                    metadata['comment count'] = apost['edge_media_to_comment']['count']
            if 'tags' in apost:
                metadata['tags'] = apost['tags']
                     
            metadata['time'] = datetime.datetime.fromtimestamp(int(apost['taken_at_timestamp'])).strftime('%Y-%m-%d %H:%M')
            metadata['display_url'] = apost['display_url']
            if apost['is_video'] == True:
                metadata['is_video'] = "Video"
      
            row = []
            for anitem in metadata:
                row.append(metadata[anitem])
            csvwriter.writerow(row)
             
    outputfile.close()

In [0]:
import pandas as pd
import numpy as np
import seaborn as sns

In [0]:
forddata = 'drive/My Drive/Colab Notebooks/Instagram/ford.csv'
fordDF = pd.read_csv(forddata)

In [0]:
volvodata = 'drive/My Drive/Colab Notebooks/Instagram/volvo.csv'
volvoDF = pd.read_csv(volvodata)

In [0]:
dodgedata = 'drive/My Drive/Colab Notebooks/Instagram/dodgeofficial.csv'
dodgeDF = pd.read_csv(dodgedata)

In [0]:
chevroletdata = 'drive/My Drive/Colab Notebooks/Instagram/chevrolet.csv'
chevroletDF = pd.read_csv(forddata)

In [0]:
hondadata = 'drive/My Drive/Colab Notebooks/Instagram/honda.csv'
hondaDF = pd.read_csv(hondadata)

# Ford Calculations

In [0]:
fordDF

is_video  ...              time
0      Photo  ...  2020-02-13 17:31
1      Photo  ...  2020-02-07 17:00
2      Photo  ...  2020-02-06 16:54
3      Photo  ...  2020-02-05 17:00
4      Photo  ...  2020-02-04 16:27
..       ...  ...               ...
195    Photo  ...  2017-03-07 20:01
196    Photo  ...  2017-03-07 17:11
197    Photo  ...  2017-03-07 14:03
198    Photo  ...  2017-03-06 18:30
199    Photo  ...  2017-03-01 17:44

[200 rows x 7 columns]

In [0]:
fordfollowers = parsedjsonfordprofile['GraphProfileInfo']['info']['followers_count']

In [0]:
for i in fordDF:
  fordDF['Post Engagement'] = (sum(fordDF['like count'])+ sum(fordDF['comment count']))/ fordfollowers

In [0]:
import statistics

In [0]:
fordavgpostengagement = statistics.mean(fordDF['Post Engagement'])

# Volvo Calculations

In [0]:
volvoDF

is_video  ...              time
0      Photo  ...  2020-02-14 09:22
1      Video  ...  2020-02-12 09:40
2      Photo  ...  2020-02-05 10:19
3      Photo  ...  2020-01-29 12:15
4      Photo  ...  2020-01-22 09:32
..       ...  ...               ...
195    Photo  ...  2018-09-18 09:19
196    Photo  ...  2018-09-14 07:00
197    Video  ...  2018-09-13 08:41
198    Video  ...  2018-09-05 10:50
199    Video  ...  2018-09-03 07:02

[200 rows x 7 columns]

In [0]:
volvofollowers = parsedjsonvolvoprofile['GraphProfileInfo']['info']['followers_count']

In [0]:
for i in volvoDF:
  volvoDF['Post Engagement'] = (sum(volvoDF['like count'])+ sum(volvoDF['comment count']))/ volvofollowers

In [0]:
volvoavgpostengagement = statistics.mean(volvoDF['Post Engagement'])

#Dodge Calculations

In [0]:
dodgefollowers = parsedjsondodgeprofile['GraphProfileInfo']['info']['followers_count']

In [0]:
for i in dodgeDF:
  dodgeDF['Post Engagement'] = (sum(dodgeDF['like count'])+ sum(dodgeDF['comment count']))/ dodgefollowers

In [0]:
dodgeavgpostengagement = statistics.mean(dodgeDF['Post Engagement'])

# Chevrolet Calculations


In [0]:
chevroletfollowers = parsedjsonchevroletprofile['GraphProfileInfo']['info']['followers_count']

In [0]:
for i in chevroletDF:
  chevroletDF['Post Engagement'] = (sum(chevroletDF['like count'])+ sum(chevroletDF['comment count']))/ chevroletfollowers

In [0]:
chevroletavgpostengagement = statistics.mean(chevroletDF['Post Engagement'])

# Honda Calculations

In [0]:
hondafollowers = parsedjsonhondaprofile['GraphProfileInfo']['info']['followers_count']

In [0]:
for i in hondaDF:
  hondaDF['Post Engagement'] = (sum(hondaDF['like count'])+ sum(hondaDF['comment count']))/ hondafollowers

In [0]:
hondaavgpostengagement = statistics.mean(hondaDF['Post Engagement'])

# Average Post Engagement Rates


In [0]:
fordavgpostengagement

2.6765647186983426

In [0]:
volvoavgpostengagement

2.7378891257542857

In [0]:
dodgeavgpostengagement

4.175391908631421

In [0]:
chevroletavgpostengagement

3.4616356466851133

In [0]:
hondaavgpostengagement

0.8488182071242392

Dodge has the highest average post engagement rate.
Honda's engagement rate is the lowest compared to its competitors. Honda's engagement rate is 0.85 while the others' rates do not go below 2.5.